### PR TITLE
Remove legacy session service

### PR DESCRIPTION
**Pull request by xacrimon**
Paired with https://github.com/gravitational/teleport.e/pull/504

Removes the legacy session service since Moderated Sessions trackers are now used instead. All dependent usage on this service was phased out in v10 so I believe should be clear to do this.

### CODE REVIEW

**Review by nklaassen — COMMENTED** [review]
Looks okay but `TestIntegrations/SessionRecordingModes/StrictMode/BeforeStartFailure` is failing

File: lib/session/session.go [hunk 1]
```diff
@@ -19,21 +19,15 @@ limitations under the License.
...
 import (
-	"context"
-	"encoding/json"
 	"fmt"
-	"sort"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/google/uuid"
```
Comment (espadolini) [inline lib/session/session.go:27]: goimports? 🙏

**Review by espadolini — APPROVED** [review]

**Comment by zmb3** [comment]
Looks like audit on is flaky:
```
--- FAIL: TestIntegrations (52.44s)
    --- FAIL: TestIntegrations/AuditOn (52.43s)
        --- FAIL: TestIntegrations/AuditOn/recording_proxy (14.18s)
            integration_test.go:379: Skipping mismatching session ed8d74c8-eec4-459f-ae7f-1fbd11f21a08, expecting upload of 63fd3d91-7e47-4db5-abe3-67a6a4130dec.
            integration_test.go:379: Skipping mismatching session ed8d74c8-eec4-459f-ae7f-1fbd11f21a08, expecting upload of 63fd3d91-7e47-4db5-abe3-67a6a4130dec.
            integration_test.go:385: recording proxy: Timeout waiting for upload of session 63fd3d91-7e47-4db5-abe3-67a6a4130dec to complete to
        --- FAIL: TestIntegrations/AuditOn/recording_proxy,_sync_recording (14.15s)
            integration_test.go:379: Skipping mismatching session eae5b3a8-20c3-40d8-8c76-a0c0afa3955c, expecting upload of c89f61af-5399-4709-b126-ff3aa17ed65b.
            integration_test.go:385: recording proxy, sync recording: Timeout waiting for upload of session c89f61af-5399-4709-b126-ff3aa17ed65b to complete to
FAIL
FAIL	github.com/gravitational/teleport/integration	53.607s
FAIL
```

**Comment by xacrimon** [comment]
@nklaassen That was flaky, I took a look and it doesn't look like my code change is related to anything in that test.

**Comment by xacrimon** [comment]
> Looks like audit on is flaky:
> 
> ```
> --- FAIL: TestIntegrations (52.44s)
>     --- FAIL: TestIntegrations/AuditOn (52.43s)
>         --- FAIL: TestIntegrations/AuditOn/recording_proxy (14.18s)
>             integration_test.go:379: Skipping mismatching session ed8d74c8-eec4-459f-ae7f-1fbd11f21a08, expecting upload of 63fd3d91-7e47-4db5-abe3-67a6a4130dec.
>             integration_test.go:379: Skipping mismatching session ed8d74c8-eec4-459f-ae7f-1fbd11f21a08, expecting upload of 63fd3d91-7e47-4db5-abe3-67a6a4130dec.
>             integration_test.go:385: recording proxy: Timeout waiting for upload of session 63fd3d91-7e47-4db5-abe3-67a6a4130dec to complete to
>         --- FAIL: TestIntegrations/AuditOn/recording_proxy,_sync_recording (14.15s)
>             integration_test.go:379: Skipping mismatching session eae5b3a8-20c3-40d8-8c76-a0c0afa3955c, expecting upload of c89f61af-5399-4709-b126-ff3aa17ed65b.
>             integration_test.go:385: recording proxy, sync recording: Timeout waiting for upload of session c89f61af-5399-4709-b126-ff3aa17ed65b to complete to
> FAIL
> FAIL	github.com/gravitational/teleport/integration	53.607s
> FAIL
> ```

@zmb3 It indeed does, but it looks to be related to the upload completer being late and not any of the timing-related fetching code changed in this PR as it errors on a seperate timeout that is only started after the changes made here execute.
```go
			// wait for the upload of the right session to complete
			timeoutC := time.After(10 * time.Second)
		loop:
			for {
				select {
				case event := <-teleport.UploadEventsC:
					if event.SessionID != tracker.GetSessionID() {
						t.Logf("Skipping mismatching session %v, expecting upload of %v.", event.SessionID, tracker.GetSessionID())
						continue
					}
					break loop
				case <-timeoutC:
					dumpGoroutineProfile()
					t.Fatalf("%s: Timeout waiting for upload of session %v to complete to %v",
						tt.comment, tracker.GetSessionID(), tt.auditSessionsURI)
				}
			}
			```

**Comment by zmb3** [comment]
>  it looks to be related to the upload completer being late and not any of the timing-related fetching code changed in this PR as it errors on a seperate timeout that is only started after the changes made here execute. So I'd deem this out of scope.

Sure, but the upload completer is based on the session tracker system and this PR fully cuts us over to the session tracker system, so it seems related enough to fix here unless it's a major effort.

**Comment by xacrimon** [comment]
@zmb3 Flake fixed.

**Comment by zmb3** [comment]
These could be unrelated, but `TestClientDisconnect` (which was flaky back in 2021 but then fixed) is failing:

```
===================================================
OUTPUT github.com/gravitational/teleport/lib/srv/regular.TestClientDisconnect
===================================================
=== RUN   TestClientDisconnect
=== PAUSE TestClientDisconnect
=== CONT  TestClientDisconnect
=== CONT  TestClientDisconnect
    sshserver_test.go:225: 
        	Error Trace:	sshserver_test.go:225
        	            				sshserver_test.go:117
        	            				sshserver_test.go:1398
        	Error:      	Received unexpected error:
        	            	timeout waiting for announce to be sent
        	Test:       	TestClientDisconnect
```

Similar timeout for `TestAdvertiseAddr`:

```
===================================================
OUTPUT github.com/gravitational/teleport/lib/srv/regular.TestAdvertiseAddr
===================================================
=== RUN   TestAdvertiseAddr
=== PAUSE TestAdvertiseAddr
=== CONT  TestAdvertiseAddr
=== CONT  TestAdvertiseAddr
    sshserver_test.go:225: 
        	Error Trace:	sshserver_test.go:225
        	            				sshserver_test.go:111
        	            				sshserver_test.go:482
        	Error:      	Received unexpected error:
        	            	timeout waiting for announce to be sent
        	Test:       	TestAdvertiseAddr
```

**Comment by xacrimon** [comment]
@zmb3 Those seem completely unrelated but we should probably fix them in another PR.

**Review by nklaassen — APPROVED** [review]

**Pull request merged**